contracts/VM.sol

```diff
@@ -41,43 +43,57 @@ abstract contract VM {
         bytes memory outdata;
 
         uint256 commandsLength = commands.length;
-        for (uint256 i; i < commandsLength; i=_uncheckedIncrement(i)) {
+        for (uint256 i; i < commandsLength; i = _uncheckedIncrement(i)) {
             command = commands[i];
             flags = uint256(uint8(bytes1(command << 32)));
 
             if (flags & FLAG_EXTENDED_COMMAND != 0) {
-                indices = commands[i++];
+                indices = commands[_uncheckedIncrement(i)];
             } else {
                 indices = bytes32(uint256(command << 40) | SHORT_COMMAND_FILL);
             }
 
             if (flags & FLAG_CT_MASK == FLAG_CT_DELEGATECALL) {
-                (success, outdata) = address(uint160(uint256(command))).delegatecall( // target
-                    // inputs
-                    state.buildInputs(
-                        //selector
-                        bytes4(command),
-                        indices
-                    )
-                );
+                (success, outdata) = address(uint160(uint256(command))) // target
+                    .delegatecall(
+                        // inputs
+                        flags & FLAG_FALLBACK == 0
+                            ? state.buildInputs(
+                                bytes4(command), // selector
+                                indices
+                            )
+                            : state[
+                                uint8(bytes1(indices)) &
+                                    CommandBuilder.IDX_VALUE_MASK
+                            ]
+                    );
             } else if (flags & FLAG_CT_MASK == FLAG_CT_CALL) {
                 (success, outdata) = address(uint160(uint256(command))).call( // target
                     // inputs
-                    state.buildInputs(
-                        //selector
-                        bytes4(command),
-                        indices
-                    )
+                    flags & FLAG_FALLBACK == 0
+                        ? state.buildInputs(
```

Comment:
I've found anecdotally that the ternary operator actually has higher gas cost than unrolling to  "if else"